### ISSUE TITLE
Downport to 700

### ISSUE THREAD

**Issue by larshp** [larshp/abapGit#635]
This question comes up every now and then, so lets add an issue for it and discuss here.

Main problem that I have found is that the source code based class editor is not part of 700. So this would have to be implemented in abapGit: take the code from all the class includes and merge it together in 1 source file. Also guess the deserialization would have to take the merged file and split it into the appropriate includes for the methods etc.

Another problem: it is difficult to find 700 systems to be used for testing/development, as most customers run more recent versions

**Comment by larshp** [comment]
also see https://github.com/larshp/abaplint/issues/147

**Comment by peterlangner** [comment]
I have a 700 system and I have own code in it. So I am interested in this feature too. 

BUT better would be to maintain the code in a higher system version, where I can use ADT and other features to better control the code quality. The only thing I need is the possibility to check, if the code is 700 compatable, while I am coding.

What to use? Remote syntax check against the system I have? (to be honest I would love to get rid of it.) Using abaplint and code in VS Studio?

What is your opinion / advice?

**Comment by larshp** [comment]
Follow https://github.com/larshp/abaplint/issues/147 and https://github.com/larshp/abaplint/issues/91, ideally abapGit will be developed in 750+ ABAP language and automagically downported. However I dont think abapGit will be downported to 700 as there is too many dependencies to SAP standard, that does not work on 700.

**Comment by peterlangner** [comment]
cross compiling - is this a subject to abaplint?

**Comment by larshp** [comment]
some day perhaps, but its after downport and syntax checking

**Comment by Dave94** [comment]
When I started to work with abapGit I had this desire for it to support abap 700 but now I think it is unnecessary. Clients are upgrading massivly to Hana systems and it's just a matter of time (I hope not that long) for the abap 700 to be forgotten. Support for abap 700 probably would mean that cool features didn't work as they do now. I did some compatibility fixes for our's company solutions and a large part of them included disabling some features because it was impossible to reimplement them for abap700. Another disadvantage was a performance drop.

TL;DR I know it would be great to have abapGit working on every system but it would cost much work and in the result it could be a huge disappointment.

**Comment by larshp** [comment]
out of scope, closing
